pkg/pillar/cmd/loguploader/loguploader.go

```diff
@@ -408,6 +433,50 @@ func handleDNSDelete(ctxArg interface{}, key string, statusArg interface{}) {
 	log.Tracef("handleDNSDelete done for %s", key)
 }
 
+// periodically check if the log for current app metric-map url list
+// is still valid. if we don't have this domain anymore, remove them.
+// using subAppInstConfig deletion won't work since the entries of those
```

Comment:
Sure, we can have queued app logs not yet sent.
But what does a controller do with app logs it receives after the app instance has been deleted? Doesn't it drop them on the floor?

If so should we also clean up the app log files when the app instance is deleted?

Currently you delete the URLmetrics, only to have them be recreated for that app instance should there be something more to send. That seems rather confusing.

Reply:
I would think the controller side if can not find the apps, will ignore those entries, today this device sends 45k, most of them can not find.
we can clean up twice, first is after the app is deleted, then wait up to an hour to do the regular cleaning. Since the entries after this patch is small, so I would think we should not bother the first (after app del) cleanup. Let controller to ignore them is ok.

Reply:
But it would be more consistent if you deferred the cleanup while there are app logs not yet uploaded for the UUID.

Reply:
ok, that means search for the app-upload directory to see if such an app exist still, skip the cleanup if it still exists